extensions/http-curl/client/HTTPClient.cpp

```diff
@@ -87,26 +89,17 @@ HTTPClient::HTTPClient()
 }
 
 HTTPClient::~HTTPClient() {
-  if (nullptr != headers_) {
-    curl_slist_free_all(headers_);
-    headers_ = nullptr;
-  }
-  if (http_session_ != nullptr) {
-    curl_easy_cleanup(http_session_);
-    http_session_ = nullptr;
-  }
+  forceClose();
```

Comment:
Why did you remove the destruction of http_session_ ? That means we can't reclaim resources and in some cases may result in a TCP socket in CLOSE_WAIT